lua/starfall/libs_sh/convar.lua

```diff
@@ -0,0 +1,125 @@
+
+local checkluatype = SF.CheckLuaType
+local PLY_META = FindMetaTable("Player")
+
+if CLIENT then
+	SF.Permissions.registerPrivilege("convar", "Read ConVars", "Allows Starfall to read your game settings", { client = {} })
+end
+
+
+--- ConVar library https://wiki.facepunch.com/gmod/ConVar
+-- @name convar
+-- @class library
+-- @libtbl convar_library
+SF.RegisterLibrary("convar")
+
+return function(instance)
+local checkpermission = instance.player ~= SF.Superuser and SF.Permissions.check or function() end
+local Ply_GetInfo = PLY_META.GetInfo
+
+local convar_library = instance.Libraries.convar
+
+if CLIENT then
+
+	local function getValidConVar(name)
+		checkpermission(instance, nil, "convar")
+		checkluatype(name, TYPE_STRING)
+
+		local cvar = GetConVar(name)
+		if not cvar then SF.Throw("Trying to access non-existent ConVar", 3) end
+		return cvar
+	end
+
+
+	--- Check if the given ConVar exists
+	-- @param string name Name of the ConVar
```

Comment:
All these client functions need `--@client`. Would also prefer if they weren't tabbed. I think it's fine after this change.